lib/connections/service.go

```diff
@@ -237,6 +224,25 @@ next:
 	}
 }
 
+// If rate limiting is set, and based on the address we
+// limit the connection, then we wrap it in a limiter.
+func (s *Service) setupReaderAndWriterWithOptionalRatelimit(c IntermediateConnection, writeRateLimit, readRateLimit *ratelimit.Bucket, options config.OptionsConfiguration) (io.Reader, io.Writer) {
```

Comment:
Convention for utility methods that don't need the receiver is to just be regular unexported functions, usually living somewhere close underneath all the methods.